scheduler/plugin/plugin.go

```diff
@@ -191,21 +191,40 @@ func (pl *wasmPlugin) EventsToRegister() (clusterEvents []framework.ClusterEvent
 var _ framework.PreFilterExtensions = (*wasmPlugin)(nil)
 
 // AddPod implements the same method as documented on framework.PreFilterExtensions.
-func (pl *wasmPlugin) AddPod(ctx context.Context, state *framework.CycleState, podToSchedule *v1.Pod, podInfoToAdd *framework.PodInfo, nodeInfo *framework.NodeInfo) *framework.Status {
+func (pl *wasmPlugin) AddPod(ctx context.Context, state *framework.CycleState, podToSchedule *v1.Pod, podInfoToAdd *framework.PodInfo, nodeInfo *framework.NodeInfo) (status *framework.Status) {
 	// We implement PreFilterExtensions with FilterPlugin, even when the guest doesn't.
 	if pl.guestInterfaces&iPreFilterExtensions == 0 {
 		return nil // unimplemented
 	}
-	panic("TODO: scheduling: AddPod")
+
+	// Add the stack to the go context so that the corresponding host function
+	// can look them up.
+	params := &stack{pod: podToSchedule, podInfoToAdd: podInfoToAdd, nodeInfo: nodeInfo}
```

Comment:
```suggestion
	params := &stack{pod: podToSchedule, podInfoToAdd: podInfoToAdd.Pod, nodeInfo: nodeInfo.Node()}
```

I hope this works fine..
BTW `marshalIfUnderLimit` supports Pod and Node, but doesn't support PodInfo and NodeInfo.
So passing pod and node is easier.
https://github.com/kubernetes-sigs/kube-scheduler-wasm-extension/blob/b32c9665273320841c76eddd75876b51d856a706/scheduler/plugin/mem.go#L31

Reply:
Thank you so much... :pray: I passed `Pod` and `Node` directory but still the error remains...

Reply:
@chansuke 
Left several comments. Could you fix it?

Reply:
Also debugging the reason why test fails is very useful. I would debug the reason then you can easily improve your code!